src/lib/realmdb/__tests__/UserProfileStorage.js

```diff
@@ -84,13 +84,21 @@ describe('UserProfileStorage', () => {
 
   it('should save profile to the db', async () => {
     await userProfileStorage.setProfile(profile)
+    const { user_id, _id, ...fields } = await userProfileStorage.profiledb.getProfile()
 
-    const encryptedProfile = await userProfileStorage.profiledb.getProfile()
-    const { user_id, _id, ...fields } = encryptedProfile
+    expect(user_id).not.toBeNull()
+    expect(_id).not.toBeNull()
+    Object.keys(fields).forEach(key => {
+      expect(profile[key]).toEqual(fields[key].display)
```

Comment:
@julianpolcode @lukasz-kilaszewski 
that's a bug i've also commented in lukasz PR
the display should be according to the field default privacy.